Update section 2 (#400)
rancher/rke2: docs/cis_self_assessment.md

</details>

**Result:** Pass

**Audit**
Run the below command on the master node.

```bash
grep 'client-cert-auth' /var/lib/rancher/rke2/server/db/etcd/config
```

Verify that the `client-cert-auth` field is set to true.

**Remediation:**
By default, RKE2 uses a config file for etcd that can be found at `/var/lib/rancher/rke2/server/db/etcd/config`. `client-cert-auth` is set to true. No manual remediation needed. 

#### 2.3
Ensure that the `auto-tls` field is not set to `true` (Scored)
<details>
<summary>Rationale</summary>
etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should not be available to unauthenticated clients. You should enable the client authentication via valid certificates to secure the access to the etcd service.
</details>

**Result:** Pass

**Audit**
Run the below command on the master node.

```bash
grep 'auto-tls' /var/lib/rancher/rke2/server/db/etcd/config
```

Verify that if the `auto-tls` field does not exist.

**Remediation:**

> By default, RKE2 uses a config file for etcd that can be found `/var/lib/rancher/rke2/server/db/etcd/config`. Within the file, it does not contain the `auto-tls` argument. No manual remediation needed.

Reviewer: ```suggestion
By default, RKE2 uses a config file for etcd that can be found at `/var/lib/rancher/rke2/server/db/etcd/config`. Within the file, it does not contain the `auto-tls` argument. No manual remediation needed.
```

#### 2.4
Ensure that the `peer-cert-file` and `peer-key-file` fields are set as appropriate (Scored)
<details>
<summary>Rationale</summary>
etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should be encrypted in transit and also amongst peers in the etcd clusters.
</details>

**Result:** *Pass

**Audit**
Run the below command on the master node.

```bash
grep -E 'peer-server-client.crt|peer-server-client.key' /var/lib/rancher/rke2/server/db/etcd/config
```

Verify that the `peer-server-client.crt` and `peer-server-client.key` fields are set as appropriate.

**Remediation:**

> By default, RKE2 uses a config file for etcd that can be found `/var/lib/rancher/rke2/server/db/etcd/config`. Within the file, the `peer-server-client.crt` and `peer-server-client.key` fields are set. No manual remediation needed.

Reviewer: ```suggestion
By default, RKE2 uses a config file for etcd that can be found at `/var/lib/rancher/rke2/server/db/etcd/config`. Within the file, the `peer-server-client.crt` and `peer-server-client.key` fields are set. No manual remediation needed.
```

#### 2.5
Ensure that the `client-cert-auth` field is set to `true` (Scored)
<details>
<summary>Rationale</summary>
etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should be accessible only by authenticated etcd peers in the etcd cluster.
</details>

**Result:** **Not Applicable**

**Audit**
Run the below command on the master node.

```bash
grep 'client-cert-auth' /var/lib/rancher/rke2/server/db/etcd/config
```

Verify that the `client-cert-auth` field in the peer section is set to true.

**Remediation:**

> By default, RKE2 uses a config file for etcd that can be found `/var/lib/rancher/rke2/server/db/etcd/config`. Within the file, the `client-cert-auth` field is set. No manual remediation needed.

Reviewer: ```suggestion
By default, RKE2 uses a config file for etcd that can be found at `/var/lib/rancher/rke2/server/db/etcd/config`. Within the file, the `client-cert-auth` field is set. No manual remediation needed.
```

#### 2.6
Ensure that the `peer-auto-tls` field is not set to `true` (Scored)
<details>
<summary>Rationale</summary>
etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should be accessible only by authenticated etcd peers in the etcd cluster. Hence, do not use self- signed certificates for authentication.
</details>

**Result:** Pass

**Audit**
Run the below command on the master node.

```bash
grep 'peer-auto-tls' /var/lib/rancher/rke2/server/db/etcd/config
```

Verify that if the `peer-auto-tls` field does not exist.

**Remediation:**

> By default, RKE2 uses a config file for etcd that can be found `/var/lib/rancher/rke2/server/db/etcd/config`. Within the file, it does not contain the `peer-auto-tls` field. No manual remediation needed.

Reviewer: ```suggestion
By default, RKE2 uses a config file for etcd that can be found at `/var/lib/rancher/rke2/server/db/etcd/config`. Within the file, it does not contain the `peer-auto-tls` field. No manual remediation needed.
```

etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should be encrypted in transit.
</details>

> **Result:** **Not Applicable**

Reviewer: Let's mark this as Pass and update the audit and remediation to reflect that we use a config file and that is what should be checked, not command line arguments.

etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should not be available to unauthenticated clients. You should enable the client authentication via valid certificates to secure the access to the etcd service.
</details>

> **Result:** **Not Applicable**

Reviewer: same as 2.1

etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should not be available to unauthenticated clients. You should enable the client authentication via valid certificates to secure the access to the etcd service.
</details>

> **Result:** **Not Applicable**

Reviewer: same

etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should be encrypted in transit and also amongst peers in the etcd clusters.
</details>

> **Result:** **Not Applicable**

Reviewer: same

etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should be accessible only by authenticated etcd peers in the etcd cluster.
</details>

> **Result:** **Not Applicable**

Reviewer: you guessed it: same

etcd is a highly-available key value store used by Kubernetes deployments for persistent storage of all of its REST API objects. These objects are sensitive in nature and should be accessible only by authenticated etcd peers in the etcd cluster. Hence, do not use self- signed certificates for authentication.
</details>

> **Result:** **Not Applicable**

Reviewer: same

Authentication to etcd is based on whether the certificate presented was issued by a trusted certificate authority. There is no checking of certificate attributes such as common name or subject alternative name. As such, if any attackers were able to gain access to any certificate issued by the trusted certificate authority, they would be able to gain full access to the etcd database.
</details>

> **Result:** **Not Applicable**

Reviewer: Same. We pass this.

**Audit**
Run the below command on the master node.

```bash
/bin/ps -ef | /bin/grep etcd | /bin/grep -v grep

> ```

Reviewer: Audit is incomplete...you also need to find the apiserver process to get the path to its ca file: 
```
ps -ef | grep apiserver | /bin/grep -v grep
```
That maps to this statement in the verification "...by the `--client-ca-file` for apiserver"

> grep 'trusted-ca-file' /var/lib/rancher/rke2/server/db/etcd/config

Reviewer: ```suggestion
# To find the ca file used by etcd:
grep 'trusted-ca-file' /var/lib/rancher/rke2/server/db/etcd/config
# To find the apiserver process:
ps -ef | grep apiserver | grep -v grep
```

> Verify that the file referenced by the `--client-ca-file` forapiserver is different from the `--trusted-ca-file` used by etcd.

Reviewer: forapiserver - looks like a pdf copy paste issue

> Verify that the file referenced by the `client-ca-file` for apiserver is different from the `trusted-ca-file` used by etcd.

Reviewer: ```suggestion
Verify that the file referenced by the `client-ca-file` flag in the apiserver process is different from the file referenced by the `trusted-ca-file` parameter in the etcd configuration file.
```

**Remediation:**

> By default, RKE2 uses a config file for etcd that can be found `/var/lib/rancher/rke2/server/db/etcd/config`. The `server-client.crt` field is set. No manual remediation needed.

Reviewer: ```suggestion
By default, RKE2 uses a config file for etcd that can be found `/var/lib/rancher/rke2/server/db/etcd/config` and the `trusted-ca-file` parameters in it are set to unique values specific to etcd. No manual remediation needed.
```

Authentication to etcd is based on whether the certificate presented was issued by a trusted certificate authority. There is no checking of certificate attributes such as common name or subject alternative name. As such, if any attackers were able to gain access to any certificate issued by the trusted certificate authority, they would be able to gain full access to the etcd database.
</details>

**Result:** **Not Applicable**

**Audit**
Run the below command on the master node.

```bash
# To find the ca file used by etcd:
grep 'trusted-ca-file' /var/lib/rancher/rke2/server/db/etcd/config
# To find the kube-apiserver process:
ps -ef | grep kube-apiserver | grep -v grep
```

Verify that the file referenced by the `client-ca-file` flag in the apiserver process is different from the file referenced by the `trusted-ca-file` parameter in the etcd configuration file.

**Remediation:**

> By default, RKE2 uses a config file for etcd that can be found `/var/lib/rancher/rke2/server/db/etcd/config` and the `trusted-ca-file` parameters in it are set to unique values specific to etcd. No manual remediation needed.

Reviewer: ```suggestion
By default, RKE2 uses a config file for etcd that can be found at `/var/lib/rancher/rke2/server/db/etcd/config` and the `trusted-ca-file` parameters in it are set to unique values specific to etcd. No manual remediation needed.
```
